### PR TITLE
Updates and tests for Drupal rules

### PR DESCRIPTION
**Pull request by fubarhouse**
## Overview

This PR introduces a change to the primary Drupal nginx rules. Given changes to this rule are specific and notably sensitive give its scale, testing has also been introduced to reflect the changes.

## Implementation

* A new field in the API which is user-configurable.
* Logic is existing, but the new configurable fields allow clients to opt-in for this logic.

## How to Test

* Build the project
   ```
   make build
   ```
* Test the project
   ```
   make test
   ```
* Look for any failing tests
   ```
   ....

   Total Duration: 0.004s
   Count: 4, Failed: 0, Skipped: 0
   ```

### Before rule changes

```
...FF

Failures/Skipped:

HTTP: http://127.0.0.1:8080/tag: status:
Expected
    <int>: 404
to equal
    <int>: 200
HTTP: http://127.0.0.1:8080/tag: Body: patterns not found: [Hello world!]

Total Duration: 0.001s
Count: 5, Failed: 2, Skipped: 0
```

### After rule change:

```
.....

Total Duration: 0.005s
Count: 5, Failed: 0, Skipped: 0
```

### CODE REVIEW

File: base/etc/nginx/conf.d/location/10-block.conf [hunk 1]
```diff
@@ -3,7 +3,7 @@ location ~ (^|/)\. {
 }
 
 # Protect files and directories from prying eyes.
-location ~* \.(log|engine|inc|install|make|module|profile|po|sh|.*sql|theme|twig|tpl(\.php)?|xtmpl|yml)(~|\.sw[op]|\.bak|\.orig|\.save)?$|/(\.(?!well-known).*|Entries.*|Repository|Root|Tag|Template|composer\.(json|lock)|web\.config)$|/#.*#$|\.php(~|\.sw[op]|\.bak|\.orig|\.save)$ {
+location ~ \.(log|engine|inc|install|make|module|profile|po|sh|.*sql|.*sql\.gz|theme|twig|tpl(\.php)?|xtmpl|yml)(~|\.sw[op]|\.bak|\.orig|\.save)?$|^\/(\.(?!well-known).*|Entries.*|Repository|Root|Tag|Template)$|(composer\.(json|lock)|web\.config|yarn\.lock|package(-lock)?\.json)$|^\/#.*#$|\.php(~|\.sw[op]|\.bak|\.orig|\.save)$ {
```
Comment (acbramley) [inline base/etc/nginx/conf.d/location/10-block.conf]: `~*` means case insensitive matching while `~` is case sensitive. Imo it should be `~*` so we continue to block `.PHP` etc...

Also, is it worth breaking up this giant regex? It's really hard to figure out what's going on, evident from the bug 😅

Comment (nickschuch) [reply base/etc/nginx/conf.d/location/10-block.conf]: Giant regex comes from Drupal.

https://git.drupalcode.org/project/drupal/-/blob/11.x/.htaccess?ref_type=heads#L6

I would prefer we keep the string as close as possible so we can copy/paste to keep up to date. Looks like `log` is the outlier. @fubarhouse can you please confirm that and break out the things that aren't Drupal into their own rules. Also comment and link to Drupal core as the reference.

Drupal cores rule is case sensitive which is why we are seeing different behaviour on our Apache based clients (hosted on other solution) vs Skpr.

Comment (acbramley) [reply base/etc/nginx/conf.d/location/10-block.conf]: So we're going to allow `.PHP` files?

Comment (fubarhouse) [reply base/etc/nginx/conf.d/location/10-block.conf]: I've confirmed that requests to URI's to `.PHP` files will not serve but will instead incur a `403` and provided a test for this.

File: base/etc/nginx/conf.d/location/10-block.conf [hunk 2]
```diff
@@ -2,8 +2,28 @@ location ~ (^|/)\. {
   return 403;
 }
 
-# Protect files and directories from prying eyes.
-location ~* \.(log|engine|inc|install|make|module|profile|po|sh|.*sql|theme|twig|tpl(\.php)?|xtmpl|yml)(~|\.sw[op]|\.bak|\.orig|\.save)?$|/(\.(?!well-known).*|Entries.*|Repository|Root|Tag|Template|composer\.(json|lock)|web\.config)$|/#.*#$|\.php(~|\.sw[op]|\.bak|\.orig|\.save)$ {
+# Protect Drupal-specific files and directories from prying eyes.
```
Comment (nickschuch) [inline base/etc/nginx/conf.d/location/10-block.conf]: I would like to keep the rules consistent with what is in the Drupal .htaccess file.

That will allow us to better maintain and keep this up to date with core.

I would only like things to be split out which are not in the Drupal .htaccess file.

Comment (fubarhouse) [reply base/etc/nginx/conf.d/location/10-block.conf]: Updated and added a before and after to the PR description to show the change enforced.

File: base/etc/nginx/conf.d/location/10-block.conf [hunk 3]
```diff
@@ -3,7 +3,12 @@ location ~ (^|/)\. {
 }
 
 # Protect files and directories from prying eyes.
```
Comment (nickschuch) [inline base/etc/nginx/conf.d/location/10-block.conf:5]: Can you please put a note on this comment referencing the Drupal .htaccess file

**Pull request merged**